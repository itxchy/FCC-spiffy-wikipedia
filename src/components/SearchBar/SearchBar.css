.form-container {
  margin: 0 auto;
  text-align: center;
  max-width: 80rem; }

.search-input {
  font-family: 'Poiret One', cursive;
  font-weight: 800;
  color: #000;
  font-size: 2em;
  min-width: 40rem; }

.search-input:input-placeholder {
  font-size: 2em;
  color: #000; }

/* Hacky fix for styling the placeholder text in the search input */
::-webkit-input-placeholder {
  font-size: 1.3em !important;
  font-weight: 500 !important;
  color: #888 !important;
  padding-left: 1rem !important;
}

:-moz-placeholder {
  /* Firefox 18- */
  font-size: 1.3em !important;
  font-weight: 500 !important;
  color: #888 !important; }

::-moz-placeholder {
  /* Firefox 19+ */
  font-size: 1.3em !important;
  font-weight: 500 !important;
  color: #888 !important; }
