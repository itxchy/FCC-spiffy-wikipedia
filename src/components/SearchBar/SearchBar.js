import React, { Component } from 'react'
import './SearchBar.css'
import { Form, FormGroup, FormControl, Button } from 'react-bootstrap'
const { func } = React.PropTypes

class SearchBar extends Component {
  constructor (props) {
    super(props)
    this.state = {
      searchText: ''
    }
    this.handleSearchInputChange = this.handleSearchInputChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSearchInputChange (event) {
    this.setState({ searchText: event.target.value })
  }

  handleSubmit (event) {
    event.preventDefault()
    if (!this.state.searchText) {
      return
    }
    this.props.onSearchSubmit(this.state.searchText.trim())
    this.setState({ searchText: '' })
  }

  render () {
    return (
      <div className='form-container'>
        <Form inline onSubmit={this.handleSubmit}>
          <FormGroup controlId='searchInput' bsSize='large'>
            <FormControl
              type='text'
              value={this.state.searchText}
              placeholder='What are you looking for?'
              className='search-input'
              onChange={this.handleSearchInputChange}
            />
          </FormGroup>
          <Button type='submit' bsSize='large' onClick={this.handleSubmit} id='search-submit'>
            Search
          </Button>
        </Form>
        <a className='chance-link' href='https://en.wikipedia.org/wiki/Special:Random'>Take a chance!</a>
      </div>
    )
  }
}

SearchBar.propTypes = {
  onSearchSubmit: func.isRequired
}

export default SearchBar
