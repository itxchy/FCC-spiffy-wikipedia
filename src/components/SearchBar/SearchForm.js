import React from 'react'
import './SearchBar.css'
import { Form, FormGroup, FormControl, Button } from 'react-bootstrap'
const { func, string } = React.PropTypes

function SearchForm (props) {
  return (
    <Form inline onSubmit={props.handleSubmit}>
      <FormGroup controlId='searchInput' bsSize='large'>
        <FormControl
          type='text'
          value={props.searchText}
          placeholder='What are you looking for?'
          className='search-input'
          onChange={props.handleSearchInputChange}
        />
      </FormGroup>
      <Button type='submit' bsSize='large' onClick={props.handleSubmit} id='search-submit'>
        Search
      </Button>
    </Form>
  )
}

SearchForm.propTypes = {
  handleSubmit: func,
  searchText: string,
  handleSearchInputChange: func
}

export default SearchForm
