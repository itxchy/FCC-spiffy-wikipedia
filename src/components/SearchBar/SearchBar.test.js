import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import SearchBar from './SearchBar'

test('SearchBar snapshot test', () => {
  const submitHandler = jest.fn()
  const component = shallow(<SearchBar onSearchSubmit={submitHandler} />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('SeachBar should change state.searchText on every keystroke', () => {
  const submitHandler = jest.fn()
  const component = mount(<SearchBar onSearchSubmit={submitHandler} />)
  component.find('.search-input').simulate('change', { target: { value: 'search term'} })
  expect(component.state('searchText')).toEqual('search term')
})

test('SearchBar should call onSearchSubmit from props on populated form submit', () => {
  const submitHandler = jest.fn()
  const component = mount(<SearchBar onSearchSubmit={submitHandler} />)
  component.setState({ searchText: 'Prince' })
  component.find('#search-submit').simulate('click')
  expect(submitHandler).toHaveBeenCalledWith('Prince')
})

test('SearchBar should not call onSearchSubmit from props if input is empty', () => {
  const submitHandler = jest.fn()
  const component = mount(<SearchBar onSearchSubmit={submitHandler} />)
  component.find('#search-submit').simulate('click')
  expect(submitHandler).not.toHaveBeenCalled()
})
