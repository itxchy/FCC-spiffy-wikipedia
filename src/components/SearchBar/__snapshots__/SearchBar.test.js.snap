exports[`test SearchBar snapshot test 1`] = `
<div
  className="form-container">
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={false}
    inline={true}
    onSubmit={[Function]}>
    <FormGroup
      bsClass="form-group"
      bsSize="large"
      controlId="searchInput">
      <FormControl
        bsClass="form-control"
        className="search-input"
        componentClass="input"
        onChange={[Function]}
        placeholder="What are you looking for?"
        type="text"
        value="" />
    </FormGroup>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsSize="large"
      bsStyle="default"
      disabled={false}
      id="search-submit"
      onClick={[Function]}
      type="submit">
      Search
    </Button>
  </Form>
  <a
    className="chance-link"
    href="https://en.wikipedia.org/wiki/Special:Random">
    Take a chance!
  </a>
</div>
`;
